File: examples/standalone/pipelineYaml/snippets/python/complex-order-explode.py
import json

order = json.loads(input)
result = []
for item in order.get("items", []):
    result.append(json.dumps({
        "order_id": order["id"],
        "item_name": item,
        "total_amount": order["total_amount"],
        "processed_at": order["processed_at"]
    }))
return result

File: examples/standalone/pipelineYaml/snippets/java/complex-order-explode.java
com.fasterxml.jackson.databind.ObjectMapper mapper = new com.fasterxml.jackson.databind.ObjectMapper();
com.fasterxml.jackson.databind.JsonNode order = mapper.readTree(input);
for (com.fasterxml.jackson.databind.JsonNode item : order.get("items")) {
    com.fasterxml.jackson.databind.node.ObjectNode record = mapper.createObjectNode();
    record.put("order_id", order.get("id").asText());
    record.put("item_name", item.asText());
    record.put("total_amount", order.get("total_amount").asDouble());
    record.put("processed_at", order.get("processed_at").asText());
    out.collect(record.toString());
}
